planning/roadmap.md

```diff
@@ -4,93 +4,316 @@ description: Planned work for the 2024 release of Thunderbird.
+High level objectives (non-exhaustive):
+* Same tray code for Linux, Windows, and macOS.
+* Unread indicator and counter.
+* Minimize to tray.
+* Start minimized.
+* Compose action per account.
+* Open settings.
+* Quit action.
+
+### Native OS Notifications
+Priority: <span style="background-color: #FFFF99">P2</span>
+Delivery Quarter: Q4 2025
+High level objectives (non-exhaustive):
+* Quick actions to reply, delete, mark messages.
+* Pause notifications for # time.
+* Manage notifications per account.
+* Focus modes.
+
+### Compose Window
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q1 2026
+High level objectives (non-exhaustive):
+* Drop the C++ editor code.
+* Implement a modern JS editor component that can be reused in calendar, tasks, and any other text area field that needs it (signature, address book notes, etc).
+* Link previews.
+* Rich text with advanced features.
+* Markdown support and preview.
+* Native emojis support (unicode).
+* Mentions support.
+
+### Settings UI
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q1 2026
+High level objectives (non-exhaustive):
+* Modal dialog, because a settings tab doesn't make sense.
+* More sections and tabs for an intuitive discovery path, drop the never ending single pages.
+* All "�Appearanc" settings in one page, including native themes (follow system, light, dark)
+* Application settings and Account settings should be in the same place.
+* "�Labs" section for listing experiments
+
+### Tasks
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q2 2026
+High level objectives (non-exhaustive):
+* Easier onboarding! Store them in the profile and sync them if there are no supported calendars.
+* Use open specs to allow interoperability.
+* Be opinionated with what we do here.
+* Modern UI.
+* Show tasks in the calendar.
+* Kanban board.
+* Integrate into emails, address books, and other areas.
+
+### Home Space
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q3 2026
+High level objectives (non-exhaustive):
+* Stats!
+* Quick links to common actions.
+* Summarization of accounts.
+* Blog listing (feed).
+* Suggested actions.
+
+### Chat Design Kickoff
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q4 2026
+High level objectives (non-exhaustive):
+* Create design plan
+* Identify areas with technical debt
+* Prioritize existing features
+* Prioritize new features
+
+
+## Back-end Priorities
+
+### Exchange Email
+Priority: <span style="background-color: #FF8080">P1</span>
+Delivery Quarter: Q1 2025
+High level objectives (non-exhaustive):
+* Full support of email operations.
+* Full support of account creation.
+
+### Global Database
+Priority: <span style="background-color: #FF8080">P1</span>
+Delivery Quarter: Q2 2025
+High level objectives (non-exhaustive):
+* SQLite based.
+* Rust interfaces.
+* Migrations.
+* Rollbacks.
```

Comment:
I don't know what "Migrations"/"Rollbacks" mean in this context.

Does this refer to the DB as a whole, or to transactions affecting the database (eg grouping multiple atomic operations)?
Going to assume it means the former: upgrading/downgrading the app, and migrating existing data.

Obviously we need forward migrations from older databases (including mork) to newer ones.

But there's not been any policy discussion on being able to roll back to an older version from a newer version (either legacy new db). Either you need to ship builds which can roll back from a future version (hard! error prone! likely impractical!), or you backport a rollback procedure to an earlier version and make sure users know which versions support which rollbacks.